src/screens/VisualTests/VisualTests.tsx

```diff
@@ -150,104 +93,229 @@ export const VisualTestsWithoutSelectedBuildId = ({
+  projectId,
+  gitInfo,
+  storyId,
+}: VisualTestsProps) => {
+  const { addNotification } = useStorybookApi();
+
+  const {
+    hasData,
+    hasProject,
+    hasSelectedBuild,
+    lastBuildOnBranch,
+    lastBuildOnBranchIsReady,
+    lastBuildOnBranchIsSelectable,
+    selectedBuild,
+    selectedBuildMatchesGit,
+    queryError,
+    rerunQuery,
+    userCanReview,
+  } = useBuild({
+    projectId,
+    storyId,
+    gitInfo,
+    selectedBuildInfo,
+  });
+
+  const reviewState = useReview({
+    buildIsReviewable: !!selectedBuild && selectedBuild.id === lastBuildOnBranch?.id,
+    userCanReview,
+    onReviewSuccess: rerunQuery,
+    onReviewError: (err, update) => {
+      if (err instanceof Error) {
+        addNotification({
+          id: "chromatic/errorAccepting",
+          // @ts-expect-error we need a better API for not passing a link
+          link: undefined,
+          content: {
+            headline: `Failed to ${
+              update.status === ReviewTestInputStatus.Accepted ? "accept" : "unaccept"
+            } changes`,
+            subHeadline: err.message,
+          },
+          icon: {
+            name: "cross",
+            color: "red",
+          },
+        });
+      }
+    },
+  });
+
+  // Currently only used by the sidebar button to show a blue dot ("build outdated")
+  useEffect(() => setOutdated(!selectedBuildMatchesGit), [selectedBuildMatchesGit, setOutdated]);
 
   // We always set status to the next build's status, as when we change to a new story we'll see
   // the next builds
-  const testsForStatus =
-    lastBuildOnBranch &&
-    "testsForStatus" in lastBuildOnBranch &&
-    lastBuildOnBranch.testsForStatus &&
-    getFragment(FragmentStatusTestFields, lastBuildOnBranch.testsForStatus.nodes);
+  useEffect(() => {
+    // if (!lastBuildOnBranchIsSelectable) return;
 
-  const buildStatusUpdate =
-    canSwitchToLastBuildOnBranch && testsForStatus ? testsToStatusUpdate(testsForStatus) : {};
+    const testsForStatus =
+      lastBuildOnBranch &&
+      "testsForStatus" in lastBuildOnBranch &&
+      lastBuildOnBranch.testsForStatus?.nodes &&
+      getFragment(FragmentStatusTestFields, lastBuildOnBranch.testsForStatus.nodes);
 
-  useEffect(() => {
     // @ts-expect-error The return type of this function is wrong in the API, it should allow `null` values
     updateBuildStatus((state) => ({
       ...createEmptyStoryStatusUpdate(state),
-      ...buildStatusUpdate,
+      ...testsToStatusUpdate(testsForStatus || []),
     }));
-    // We use the stringified version of buildStatusUpdate to do a deep diff
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [JSON.stringify(buildStatusUpdate), updateBuildStatus]);
 
-  const shouldSwitchToLastBuildOnBranch =
-    canSwitchToLastBuildOnBranch && lastBuildOnBranchCompletedStory;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [lastBuildOnBranchIsSelectable, JSON.stringify(lastBuildOnBranch), updateBuildStatus]);
```

Comment:
Won't this send status unnecessary status updates if the `lastBuildOnBranch` changes but the status hasn't?

I would have thought this refactor shouldn't change behaviour.

Reply:
That's not what I was concerned about. I was concerned about the change from the serialized update being the dependency to the serialized build.

Reply:
Oh I see what you mean now. Hadn't considered that. I'll change it back.

Reply:
Restored it now.